### PR TITLE
Health standardization

### PR DESCRIPTION
**Pull request by EgorDinamit**
## About the Pull Request

Ports and improves(i.e. closets and windows) the following:
- Baystation12/Baystation12/pull/31131
- Baystation12/Baystation12/pull/31189
- Baystation12/Baystation12/pull/31226
- Baystation12/Baystation12/pull/31284
- Baystation12/Baystation12/pull/31289
- Baystation12/Baystation12/pull/31322
- Baystation12/Baystation12/pull/31470

## Why It's Good For The Game

I'd explain it better, but as I see it:
- Makes everything less shitcode-y.
- Kills stupid extension code.
- Allows players to break closets with strong(20+ force) weapons.

## Did you test it?

Tested pretty much everything I could, mostly closets, blob, windows, and walls.

## Changelog

:cl: EgorDinamit, SierraKomodo (BayStation 12)
tweak: Standardized health for structures, turfs, and items. In short - less shitcode.
tweak: Closets can now be broken open with any item that has more than 20 force(damage).
/:cl:

### CODE REVIEW

File: code/_helpers/maths.dm [hunk 1]
```diff
@@ -3,7 +3,8 @@
 #define ceil(x) (-round(-(x)))
 #define CEILING(x, y) ( -round(-(x) / (y)) * (y) )
 #define MULT_BY_RANDOM_COEF(VAR,LO,HI) VAR =  round((VAR * rand(LO * 100, HI * 100))/100, 0.1)
-#define PERCENT(val, max, places) round((val) / (max) * 100, !(places) || 10 ** -(places))
+/// The percentage of value in max, rounded to places: 1 = nearest 0.1 , 0 = nearest 1 , -1 = nearest 10, etc
+#define Percent(value, max, places) round((value) / (max) * 100, !(places) || 10 ** -(places))
```
Comment (Shadowtail117) [inline code/_helpers/maths.dm]: Why make this be in lowercase?

File: code/_onclick/item_attack.dm [hunk 2]
```diff
@@ -34,8 +34,10 @@ avoid code duplication. This includes items that may sometimes act as a standard
 	return
 
 /atom/movable/attackby(obj/item/W, mob/user)
-	if(!(W.item_flags & ITEM_FLAG_NO_BLUDGEON))
+	if(W.item_flags & ITEM_FLAG_NO_BLUDGEON)
 		visible_message("<span class='danger'>[src] has been hit by [user] with [W].</span>")
+		return
+	..()
```
Comment (Shadowtail117) [inline code/_onclick/item_attack.dm:40]: This directly reverses the previous behavior, making it so items that *don't* bludgeon now bludgeon. Is this intended or am I simply reading this incorrectly?

File: code/_helpers/maths.dm (hunk 1 above)
Comment (EgorDinamit) [reply code/_helpers/maths.dm]: Was looking through code and apparently, that's what they did on Bay.
Would be useful if we ever need to port something else from it.

**Review by Shadowtail117 — CHANGES_REQUESTED** [review]
Cursory glance at the files looks okay and I don't think anyone has screamed at it. Other than the 2 suggestions and the pending addition from #267 I think this is good to go.

File: code/_helpers/maths.dm (hunk 1 above)
Comment (Shadowtail117) [reply code/_helpers/maths.dm]: It's general styling that all macros be in all uppercase, but if it's for the sake of porting compatibility, I'll ask for @Cupax3's opinion.

File: code/_onclick/item_attack.dm (hunk 2 above)
Comment (EgorDinamit) [reply code/_onclick/item_attack.dm:40]: It's exactly the opposite. Previously, items WITHOUT "no_bludgeon" flag would **not** attack normally.
I simply fixed it.

Comment (Shadowtail117) [reply code/_onclick/item_attack.dm:40]: (Resolved on Discord)

File: code/_helpers/maths.dm (hunk 1 above)
Comment (Cupax3) [reply code/_helpers/maths.dm]: No need to port Bay's inconsistency too. Keep it UPPERCASE and include a comment that'll allow searches to detect the lowercase version so it's not as jarring to find as a full-on replacement, please.

Comment (EgorDinamit) [reply code/_helpers/maths.dm]: @Cupax3 reverted that change. It's uppercase again.

**Review by Shadowtail117 — DISMISSED** [review]
I will hope that it passes screamtest.

**Comment by EgorDinamit** [comment]
Commits squashed. PR has been previously approved.

**Pull request merged**